pandas/src/inference.pyx

```diff
@@ -655,22 +731,60 @@ def maybe_convert_numeric(object[:] values, set na_values,
         val = values[i]
 
         if val.__hash__ is not None and val in na_values:
+            seen_null = True
+            if _check_uint64_nan(seen_uint, seen_null,
```

Comment:
Why don't we leave as a follow-up for the time being.  I'm not sure yet about this.